codec/collections.go

```
@@ -179,3 +324,104 @@ func (c collInterfaceValue[T]) ValueType() string {
 	var t T
 	return fmt.Sprintf("%T", t)
 }
+
+// SchemaCodec returns a schema codec, which will always have a single JSON field
+// as there is no way to know in advance the necessary fields for an interface.
+func (c collInterfaceValue[T]) SchemaCodec() (collcodec.SchemaCodec[T], error) {
+	var (
```

golangci-lint (1.61.0)

File is not `gofumpt`-ed with `-extra`

(gofumpt)